src/veins/base/toolbox/Signal.cc

```diff
@@ -0,0 +1,995 @@
+}
+
+double Signal::getRelativeMax() const
+{
+	return *(std::max_element(values+relativeOffset, values+relativeOffset+numRelativeValues));
+}
+
+double Signal::getDataMax() const
+{
+	return *(std::max_element(values+dataOffset, values+dataOffset+numDataValues));
+}
+
+double Signal::getMaxInRange(size_t freqIndexLow, size_t freqIndexHigh) const
+{
+	return *(std::max_element(values+freqIndexLow, values+freqIndexHigh));
+}
+
+void Signal::print() const
+{
+	if(timingUsed)
+		std::cout << "Range: " << getReceptionStart() << " - " << getReceptionEnd() << " (" << duration << ")" << std::endl;
+
+	for(uint16_t i=getRelativeStart();i<getRelativeEnd();i++)
+	{
+		std::cout << spectrum->freqAt(i) << ":\t " << values[i] << std::endl;
+	}
+
+	std::cout << "-----------------------------------------------" << std::endl;
+}
+
+void Signal::printAbsolute() const
+{
+	std::cout << "-----------------------------------------------" << std::endl;
+
+	if(timingUsed)
+		std::cout << "Range: " << getReceptionStart() << " - " << getReceptionEnd() << " (" << duration << ")" << std::endl;
+
+	for(uint16_t i=0;i<numAbsoluteValues;i++)
+	{
+		std::cout << spectrum->freqAt(i) << ":\t " << values[i] << std::endl;
+	}
+}
+
+Signal& Signal::operator= (const double value)
+{
+	for(size_t i = 0; i < numAbsoluteValues; i++)
+	{
+		values[i] = value;
+	}
+	return *this;
+}
+
+Signal& Signal::operator= (const Signal& other)
+{
+	if(this == &other)
+		return *this;
+
+	senderModuleID = other.senderModuleID;
+	senderFromGateID = other.senderFromGateID;
+	receiverModuleID = other.receiverModuleID;
+	receiverToGateID = other.receiverToGateID;
+
+	spectrum = other.getSpectrum();
+
+	relativeOffset = other.getRelativeOffset();
+	dataOffset = other.getDataOffset();
+
+	centerFrequencyIndex = other.getCenterFrequencyIndex();
+
+	numAbsoluteValues = other.getNumAbsoluteValues();
+	numRelativeValues = other.getNumRelativeValues();
+	numDataValues = other.getNumDataValues();
+
+	//TODO: Check if new spectrum has changed and new one is larger
+	if(values)
+		delete[] values;
+	values = new double[spectrum->getNumFreqs()]{0};
+
+	std::copy(other.getAbsoluteValues(), other.getAbsoluteValues()+numAbsoluteValues, values);
+
+	analogueModelList = other.getAnalogueModelList();
+	numAnalogueModelsApplied = other.getNumAnalogueModelsApplied();
+	senderPos = other.getSenderPos();
+	receiverPos = other.getReceiverPos();
+
+	timingUsed = other.hasTiming();
+	sendingStart = other.getSendingStart();
+	duration = other.getDuration();
+	propagationDelay = other.getPropagationDelay();
+
+	bitrate = other.getBitrate();
+
+	return *this;
+}
+
+Signal& Signal::operator+= (const Signal& other)
+{
+	assert(this->getSpectrum() == other.getSpectrum());
+
+	size_t lowIndex = std::min(this->getRelativeStart(), other.getRelativeStart());
+	size_t highIndex = std::max(this->getRelativeEnd(), other.getRelativeEnd());
+
+	for(size_t i= lowIndex; i < highIndex; i++)
```

Comment:
If the spectrums are the same, aren't this->values and other.values indexed in the same way? If that is the case, they could be directly combined using `std::transform` (there are overloads for combining two ranges into one).